app/models/payment/braintree_payment_method.rb

```diff
@@ -1,3 +1,4 @@
 class Payment::BraintreePaymentMethod < ActiveRecord::Base
-  belongs_to :customer, class_name: Payment::BraintreeCustomer, foreign_key: :customer_id
+  belongs_to :customer,     class_name: Payment::BraintreeCustomer
```

Comment:
I was stubbornly insisting on keeping the customer_id FK. Neal was convincing me it might be a good idea to just go the rails' way. I guess it's a lesson for me to be less stubborn with Neal.